### PR TITLE
Add Device time of report capture to JSON payload

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Reports may be cached on a device if no network connection is available. In this case, the report may not be delivered for hours, or even days, after the error was originally captured by Bugsnag.

Sending the time at which the report was originally captured allows for timestamps to be normalised. This avoids confusing information similar to the following being shown:

screen shot 2018-10-23 at 16 19 51

## Design

The pipeline normalises timestamps based off the `device.time` field.

## Changeset

Serialises the time at which the report was generated into the `device.time` payload field.

## Tests

- Added mazerunner tests for handled/unhandled errors, payload serialisation unit test
- Checked payload manually for presence of `device.time` in handled/unhandled error
- Checked breadcrumb time is updated in the dashboard when sending in a payload that has been cached and had the timestamp manually edited

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: features/handled_errors.feature [hunk 1]
```diff
@@ -6,6 +6,7 @@ Scenario: Override errorClass and message from a notifyError() callback
     And the request is a valid for the error reporting API
     And the exception "errorClass" equals "Bar"
     And the exception "message" equals "Foo"
+    And the payload field "events.0.device.time" is not null
```
Comment (snmaynard) [inline features/handled_errors.feature]: I would like to verify the format of this field and also that it is the current time.

**Comment by fractalwrench** [comment]
I have updated the mazerunner scenario to check that the field is a timestamp. There's also some coverage in the unit tests for ensuring the timestamp is serialised from the KSCrash field, if that's sufficient?

Note: Cocoa mazerunner scenarios still do not work on my machine and this makes it very hard to write/debug scenarios.

**Comment by fractalwrench** [comment]
I have updated the mazerunner scenario to check that the field is a timestamp within 30s of the current time .

File: features/steps/ios_steps.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+require 'date'
```
Comment (snmaynard) [inline features/steps/ios_steps.rb]: You dont use this require.

**Pull request merged**